packages/orm/src/Model.ts

```diff
@@ -252,27 +253,28 @@ export default class Model {
         ].join("")
       );
     }
-    const props = { ...userProps };
+    const props: Record<string, AnyModel | ModelId | null | (AnyModel | ModelId | null)[]> = { ...userProps };
 
-    const m2mRelations: Record<string, any> = {};
+    const m2mRelations: Record<string, (AnyModel | ModelId)[]> = {} as Record<string, (AnyModel | ModelId)[]>;
 
     const declaredFieldNames = Object.keys(this.fields);
     const declaredVirtualFieldNames = Object.keys(this.virtualFields);
 
     declaredFieldNames.forEach((key) => {
-      const field = this.fields[key];
+      const fields = this.fields;
+      const field = fields[key];
       const valuePassed = userProps.hasOwnProperty(key);
       if (!(field instanceof ManyToMany)) {
         if (valuePassed) {
-          const value = userProps[key];
-          props[key] = normalizeEntity(value);
-        } else if (field.getDefault) {
-          props[key] = field.getDefault();
+          const value = (userProps[key as keyof RefWithFields<InstanceType<MClassType>>] as unknown) as AnyModel;
```

Comment:
I wrote castTo<DestT>(arg) function exactly for this kind of situations - to simplify type casts. You may use it.